arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+### Use case View
+
+For heterogeneous object management, bucket creation is important. Currently soda-multicloud supports creation of folders under cloud buckets as shown in Fig-1. So, soda-users must need to create cloud buckets manually and provide a name, AK/SK to soda for folder creation. As shown in fig-1, step-1 is manual which is performed by the user, whereas step-2,3,4,.. are supported by soda-multi-cloud.
+
+To avoid manual intervention, step-1 should also be part of soda-multicloud as shown in Fig-2. This way users need not to do anything manually on the cloud vendors console.
+
+Once the buckets are created, users should perform all the operations supported by soda-multicloud. For example: Uploading objects, setting lifecycle rules, migration etc.  
+
+
+![](resources/multi-cloud-fig1.png)
+
+
+![](resources/multi-cloud-fig2.png)
+
+
+
+#### API's(Create/Delete) to support bucket management:
+
+To support bucket managament, below is the list of API's with detailed description about parameters for various cloud vendors are below:
+
+
+#### AWS
+
+##### Rules for bucket naming:
+
+The following rules apply for naming S3 buckets:
+
+* Bucket names must be between 3 and 63 characters long.
+
+* Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
+
+* Bucket names must begin and end with a letter or number.
+
+* Bucket names must not be formatted as an IP address (for example, 192.168.5.4).
+
+* Bucket names can't begin with xn-- (for buckets created after February 2020).
+
+* Bucket names must be unique within a partition. A partition is a grouping of Regions. AWS currently has three partitions: aws (Standard Regions), aws-cn (China Regions), and aws-us-gov (AWS GovCloud [US] Regions).
+
+* Buckets used with Amazon S3 Transfer Acceleration can't have dots (.) in their names. For more information about transfer acceleration, see Amazon S3 Transfer Acceleration.
+
+
+##### URI Request Parameters
+
+The request uses the following URI parameters.
+
+`Bucket`: The name of the bucket to create.
+ Required: Yes
+
+`x-amz-acl`: The canned ACL to apply to the bucket.
+
+`Valid Values`: private | public-read | public-read-write | authenticated-read
+
+`x-amz-bucket-object-lock-enabled`: Specifies whether you want S3 Object Lock to be enabled for the new bucket.
+
+`x-amz-grant-full-control`: Allows grantee the read, write, read ACP, and write ACP permissions on the bucket.
+
+`x-amz-grant-read`: Allows grantee to list the objects in the bucket.
+
+`x-amz-grant-read-acp`: Allows grantee to read the bucket ACL.
+
+`x-amz-grant-write`: Allows grantee to create, overwrite, and delete any object in the bucket.
+
+`x-amz-grant-write-acp`: Allows grantee to write the ACL for the applicable bucket.
+
+##### API:
+***
+PUT / HTTP/1.1/BucketName
+***
+##### Request Body
+The request accepts the following data in XML format.
```

Comment:
Is there any constraint to provide requests in XML format? Can we standardize multi-cloud APIs as multi-cloud API uses different data formats for different APIs?

Reply:
Currently we are only suporting XML. Yes, can be improved to support different formats as well in coming releases.